[Follow-Up] Updating TC_ACL_2_6 to add subscription validation and fix for issue #38663

j-ororke opened the pull request:
Updating TC_ACL_2_6 to add subscription validation  for read events:
- Added additional subscription validation to test steps 3, 5, and 7
- Follow-Up for follow-up task here: #[548](https://github.com/project-chip/matter-test-scripts/issues/548)
- Adding value fix for issue [38663](https://github.com/project-chip/connectedhomeip/issues/38663)

#### Testing
- Adds additional subscription validation to test steps 3, 5, and 7

j-ororke committed:
[Follow-Up] Updating TC_ACL_2_6 to add subscription validation for read events
- Added additional subscription validation to test steps 3, 5, and 7
- Follow-Up for follow-up task here: https://github.com/project-chip/matter-test-scripts/issues/548

restyled-commits committed:
Restyled by autopep8

restyled-commits committed:
Restyled by isort

j-ororke committed:
Resolving linting errros

j-ororke committed:
Updating TC_ACL_2_6 test module:
- Changing step 7 back to remove the subscription as this step expects no new entries

cecille commented on src/python_testing/TC_ACL_2_6.py:
```diff
@@ -136,26 +144,90 @@ async def test_TC_ACL_2_6(self):
-        logging.info(f"Events response: {events_response2}")
-        logging.info(f"Event response length: {len(events_response2)}")
-        asserts.assert_true(len(events_response2) == 2, "Expected 2 events")
+        # Read events with retries
```
Since you've already got the event via a subscription, you won't need the retries here any more.

j-ororke committed:
Updating TC_ACL_2_6 python3 test module:
- Removing for retry loop in test step 5
- Moving function out of main test block and into a class function

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Updating TC_ACL_2_6 python3 test module:
- Resolving linting errors

j-ororke committed:
Adding value fix for https://github.com/project-chip/connectedhomeip/issues/38663

j-ororke committed:
Merge branch 'master' into update_TC_ACL_2_6_subscriptions

cecille commented on src/python_testing/TC_ACL_2_6.py:
```diff
@@ -136,26 +154,69 @@ async def test_TC_ACL_2_6(self):
+                                 reverse=True)[:2]
+
+        # Verify we got the expected number of events
+        asserts.assert_true(len(read_events) >= 2,
```
I think this might end up erroring weirdly if you don't have events_response.
```suggestion
        asserts.assert_true(events_response, "Did not receive a response when calling ReadEvents")
        asserts.assert_true(len(read_events) >= 2,
```

cecille replied:
uh...no guarantees as to the correctness of that code - I wrote it in a comment, bad things may happen.

j-ororke replied:
Thank you, the new assert you added appears works perfectly, validated in local setup before adding to the code.

j-ororke committed:
Update src/python_testing/TC_ACL_2_6.py

Adding new assert in to code from Cecille.

Co-authored-by: C Freeman <cecille@google.com>

j-ororke committed:
Merge branch 'master' into update_TC_ACL_2_6_subscriptions

Alami-Amine reviewed (approved):
tested locally to confirm it passes against Matter 1.0 All clusters app
